Client/mods/deathmatch/CClient.cpp

```diff
@@ -305,6 +306,75 @@ void CClient::OnWindowFocusChange(bool state)
     g_pClientGame->OnWindowFocusChange(state);
 }
 
+void CClient::OnChatboxLayoutChange()
+{
+    if (g_pClientGame)
+    {
+        CCVarsInterface*  pCVars = g_pCore->GetCVars();
+        SString           strCVarValue;
+        std::stringstream ss;
+        float             fNumber, fX, fY;
+        int               iR, iG, iB, iA;
+        CLuaArguments     chatboxItemList;
+
+        for (unsigned int i = 0; i < MAX_CHATBOX_LAYOUT_CVARS; i++)
```

Comment:
You could use an array of struct for `g_chatboxLayoutCVars`, and store the type of value next to the cvar name. For example:
```cpp
enum class DummyType {
    STRING,
    BOOLEAN,
    FLOAT1,
    FLOAT2,
    FLOAT4,
};

struct Dummy {
    const char* name;
    DummyType   type;
};
```